cmake/modules/FindTexturePacker.cmake

```diff
@@ -9,44 +9,78 @@
                                           HINTS ${_tppath})
 
-    add_executable(TexturePacker::TexturePacker IMPORTED GLOBAL)
-    set_target_properties(TexturePacker::TexturePacker PROPERTIES
-                                                       IMPORTED_LOCATION "${TEXTUREPACKER_EXECUTABLE}")
+    add_executable(TexturePacker::TexturePacker::Executable IMPORTED GLOBAL)
+    set_target_properties(TexturePacker::TexturePacker::Executable PROPERTIES
+                                          IMPORTED_LOCATION "${TEXTUREPACKER_EXECUTABLE}")
   else()
+    set(TEXTUREPACKER_EXECUTABLE "TEXTUREPACKER_EXECUTABLE-NOTFOUND")
     if(WITH_TEXTUREPACKER)
       get_filename_component(_tppath ${WITH_TEXTUREPACKER} ABSOLUTE)
       get_filename_component(_tppath ${_tppath} DIRECTORY)
       find_program(TEXTUREPACKER_EXECUTABLE NAMES "${APP_NAME_LC}-TexturePacker" TexturePacker
-                                            HINTS ${_tppath})
+                                          HINTS ${_tppath})
 
-      include(FindPackageHandleStandardArgs)
-      find_package_handle_standard_args(TexturePacker "Could not find '${APP_NAME_LC}-TexturePacker' or 'TexturePacker' executable in ${_tppath} supplied by -DWITH_TEXTUREPACKER. Make sure the executable file name matches these names!"
-                                        TEXTUREPACKER_EXECUTABLE)
-      if(TEXTUREPACKER_FOUND)
-        add_executable(TexturePacker::TexturePacker IMPORTED GLOBAL)
-        set_target_properties(TexturePacker::TexturePacker PROPERTIES
-                                                           IMPORTED_LOCATION "${TEXTUREPACKER_EXECUTABLE}")
+      # Use external TexturePacker executable if found
+      if(NOT TEXTUREPACKER_EXECUTABLE MATCHES "TEXTUREPACKER_EXECUTABLE-NOTFOUND")
+        add_executable(TexturePacker::TexturePacker::Executable IMPORTED GLOBAL)
+        set_target_properties(TexturePacker::TexturePacker::Executable PROPERTIES
+                                          IMPORTED_LOCATION "${TEXTUREPACKER_EXECUTABLE}")
+      else()
+        # Warn about external TexturePacker supplied but not fail fatally
+        # because we might have internal TexturePacker executable built
+        message(WARNING "Could not find '${APP_NAME_LC}-TexturePacker' or 'TexturePacker' executable in ${_tppath} supplied by -DWITH_TEXTUREPACKER. Make sure the executable file name matches these names!")
       endif()
-      mark_as_advanced(TEXTUREPACKER)
-    else()
+    endif()
+
+    # Ship TexturePacker only on Linux and FreeBSD
+    if(CMAKE_SYSTEM_NAME MATCHES "FreeBSD" OR CMAKE_SYSTEM_NAME MATCHES "Linux")
+      set(INTERNAL_TEXTUREPACKER_INSTALLABLE TRUE)
+    endif()
+
+    # Use it during build if build architecture is same as host
+    # (not cross-compiling) and TEXTUREPACKER_EXECUTABLE is not found
+    if(CORE_HOST_IS_TARGET AND TEXTUREPACKER_EXECUTABLE MATCHES "TEXTUREPACKER_EXECUTABLE-NOTFOUND")
```

Comment:
```suggestion
    if(CORE_HOST_IS_TARGET AND NOT TEXTUREPACKER_EXECUTABLE)
```

Reply:
OK if I reset the `TEXTUREPACKER_EXECUTABLE` on `find_program`s failure.